File: Arrays/0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.empty()) return 0;
        unordered_set<int> myset;
        for(int num : nums) myset.insert(num);
        int len = 1;
        for(int num : myset)
        {
            if(!myset.count(num - 1)){
                int cur = 0;
                int curnum = num;
                while(myset.count(curnum++)) cur++;
                len = max(len, cur);
            }
        }
        return len;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        
        myset = set(nums)
        lenght = 1
        for num in myset:
            if num-1 not in myset:
                cur = 0
                curnum = num
                while curnum in myset:
                    cur+= 1
                    curnum+= 1
                lenght = max(lenght, cur)
        return lenght
        